app/services/dashboard_service.rb

```diff
@@ -0,0 +1,61 @@
+class DashboardService
```

Comment:
Instead of calling `Transaction.all`  in each method, I would cache the result of that call in an instance variable (which means you would also have to change all methods into instance methods)